Source/ManagedObjectContext/StorageStack+Backup.swift

```
@@ -137,13 +141,26 @@ extension StorageStack {
         }
     }
     
-    private static func markAsImported(coordinator: NSPersistentStoreCoordinator, location: URL, options: [String: Any]) throws {
+    private static func markAsImported(coordinator: NSPersistentStoreCoordinator,
+                                       location: URL,
+                                       options: [String: Any],
+                                       encryptionKeys: EncryptionKeys? = nil) throws {
         // Add persistent store at the new location to allow creation of NSManagedObjectContext
         let store = try coordinator.addPersistentStore(ofType: NSSQLiteStoreType, configurationName: nil, at: location, options: options)
         let context = NSManagedObjectContext(concurrencyType: .privateQueueConcurrencyType)
         context.persistentStoreCoordinator = coordinator
         var storeMetadata = store.metadata
         
+        if let encryptionKeys = encryptionKeys {
+            context.performAndWait {
+                do {
+                    try context.disableEncryptionAtRest(encryptionKeys: encryptionKeys)
```

How about doing this only if `context.encryptMessagesAtRest` is `true`?

---

Additionally, we should probably throw an error when EAR is enabled but encryption keys were not passed to the method. There's no point creating a backup if we can't decrypt it first.

---

I will add an `throw` error for the case you mentioned but right before the start of the backup.

---

> I will add an `throw` error for the case you mentioned but right before the start of the backup.

Hmm why not do check here? Then you also don't need to expose the `getEncryptionKeys()` since then you could access the encryption keys directly on the context `uiContext.encryptionsKeys`.